core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/keystore/persistence/impl/JDBCKeyStorePersistenceManager.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.keystore.persistence.impl;
+
+import org.apache.axis2.util.UUIDGenerator;
+import org.wso2.carbon.keystore.persistence.KeyStorePersistenceManager;
+import org.wso2.carbon.keystore.persistence.dao.KeyStoreDAO;
+import org.wso2.carbon.keystore.persistence.model.KeyStoreModel;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This implementation handles the keystore storage/persistence related logics in the Database.
+ */
+public class JDBCKeyStorePersistenceManager implements KeyStorePersistenceManager {
```

Comment:
Do we need this class separating the logic from the DAO layer?

Out of the 8 public methods implemented here, 6 of them dont have any logic, rather they just call the dao directly..

Regarding the other two methods,
* For `getKeyStore` method, if we improve the KeyStorePersistenceManager to return `Optional<KeyStoreModel>`, again `getKeyStore` will have no logic.
* For `addKeystore` method, we can consider moving `isKeyStoreExistsInDatabase()` validation and other logic to the KeyStoreManager, since those logics so can be seen duplicated in the `RegistryKeyStorePersistenceManager` as well.